lab_2/main.py

```diff
@@ -34,8 +51,33 @@ def fill_lcs_matrix(first_sentence_tokens: tuple, second_sentence_tokens: tuple)
     :param second_sentence_tokens: a tuple of tokens
     :return: a lcs matrix
     """
-    pass
-
+    if not isinstance (first_sentence_tokens, tuple) or not isinstance (second_sentence_tokens, tuple):
+        return []
+    if len(first_sentence_tokens) == 0 or len(second_sentence_tokens) == 0:
+        return []
+    list_of_tuples = [first_sentence_tokens, second_sentence_tokens]
+    for current_tuple in list_of_tuples:
+        for word in current_tuple:
+            if not isinstance(word, str):
+                return []
+    matrix = create_zero_matrix(len(first_sentence_tokens) + 1, len(second_sentence_tokens) + 1)
+    new_matrix = create_zero_matrix(len(first_sentence_tokens), len(second_sentence_tokens))
+    row_counter = 0
+    for row in matrix:
+        if row_counter != 0:
+            column_counter = 0
+            while column_counter < len(row):
+                if column_counter != 0:
+                    if first_sentence_tokens[row_counter - 1] == second_sentence_tokens[column_counter - 1]:
+                        matrix[row_counter][column_counter] = matrix[row_counter - 1][column_counter - 1] + 1
+                        new_matrix[row_counter - 1][column_counter - 1] = matrix[row_counter][column_counter]
+                    else:
+                        matrix[row_counter][column_counter] = (max(matrix[row_counter][column_counter - 1],
+                            matrix[row_counter - 1][column_counter]))
```

Comment:
if row_counter == 0 or column_counter == 0, where do we look here `matrix[row_counter - 1][column_counter]` or here `matrix[row_counter][column_counter - 1]`?